components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/execution/PreIssueAccessTokenRequestBuilder.java

```diff
@@ -112,12 +121,69 @@ private Event getEvent(OAuthTokenReqMessageContext tokenMessageContext, Map<Stri
             eventBuilder.userStore(new UserStore(authorizedUser.getUserStoreDomain()));
         }
 
-        eventBuilder.accessToken(getAccessToken(tokenMessageContext, claimsToAdd));
+        // Retrieving app information here to reduce the perf overhead.
```

Comment:
IMO this comment is redundant

Reply:
Fixed 91b6ce9040c88be6f1ba11fbd755a76e26411be6